java/simulated_ble/README.md

```diff
@@ -0,0 +1,306 @@
+# Azure IoT Edge BLE Data Converter Module
+
+## Overview
+
+This tutorial showcases how one might build a module for Azure IoT Edge in Java.
+
+In this tutorial, we will walk through environment setup and how to write a [BLE](https://en.wikipedia.org/wiki/Bluetooth_Low_Energy) data converter module using the latest Azure IoT Edge packages.
+
+## Prerequisites
+
+In this section, you will setup your environment for IoT Edge module development. It applies to both *64-bit Windows* and *64-bit Linux (Ubuntu)* operating systems.
+
+First of all, the following software is required:
+
+1. The latest [Git Client](https://https://git-scm.com/downloads).
+2. The latest [**x64** JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html).
+3. The latest [Maven](https://maven.apache.org/install.html).
+
+And now it's time to open your favorite command line terminal and navigate to the working folder:
+
+1. `git clone https://github.com/Azure-Samples/iot-edge-samples.git`.
+2. `cd iot-edge-samples/java/simulated_ble`
+
+## Overall Architecture
+
+The Azure IoT Edge heavily adopts the [Von Neumann architecture](https://en.wikipedia.org/wiki/Von_Neumann_architecture). That means the whole Azure IoT Edge is a system which processes input and produces output; and each individual module is also a tiny input-processor-output subsystem. In this tutorial, we will introduce three modules:
+
+1. A module which generates a simulated [BLE](https://en.wikipedia.org/wiki/Bluetooth_Low_Energy) signal periodically
+2. A module which converts the received [BLE](https://en.wikipedia.org/wiki/Bluetooth_Low_Energy) signal to a formatted [JSON](https://en.wikipedia.org/wiki/JSON) message
+3. A module which prints the received [JSON](https://en.wikipedia.org/wiki/JSON) message
+
+So in this case, a typical end-to-end dataflow is illustrated below:
+
+![Dataflow between three modules](dataflow.png "Input: Simulated BLE Module; Processor: Converter Module; Output: Printer Module")
+
+
+## Understand the Code
+
+### Maven Project Structure
+
+Since IoT Edge packages are based on Maven, we need to create a typical Maven project structure, which contains a `pom.xml` file.
+
+Ths POM inherits from the `com.microsoft.azure.gateway.gateway-module-base` package which declares all the dependencies needed by a module project like the runtime binaries, the gateway configuration file as well as the execution behavior. This will save us from writing hundreds of lines of code again and again!
+
+The only essential thing we need to do in our own `pom.xml` is to re-declare the dependencies and plugins used in parent, and to specify the name of the IoT Edge configuration file. Everything else is optional to module development.
```

Comment:
The only thing that is required of us is to change our `pom.xml` file, we need to re-declare the dependencies and plugins used in parent class, and to specify the name of the Azure IoT Edge configuration file. Everything else is optional and not required for module development.